src/backend/distributed/planner/merge_planner.c

```diff
@@ -1124,6 +1116,26 @@ DeferErrorIfRoutableMergeNotSupported(Query *query, List *rangeTableList,
 								"repartitioning")));
 		return deferredError;
 	}
+
+
+	/*
+	 * If execution has reached this point, it indicates that the query can be delegated to the worker.
+	 * However, before proceeding with this delegation, we need to confirm that the user is utilizing
+	 * the distribution column of the source table in the Insert variable.
+	 * If this is not the case, we should refrain from pushing down the query.
+	 */
+
+	Var *insertVar =
+		FetchAndValidateInsertVarIfExists(targetRelationId, query);
+	if (insertVar &&
+		!IsDistributionColumnInMergeSource((Expr *) insertVar, query, true))
+	{
+		ereport(DEBUG1, (errmsg(
```

Comment:
Modify this error message to say, non-distribution column is detected in Merge join condition, so the push-down will not be attempted.

Reply:
No , we are not checking join condition here right ?

We are checking that if we are inserting value of  source dist column in target dist column 
If yes then we push down otherwise we don't push down.

Reply:
Done